src/libraries/System.Net.Http/tests/FunctionalTests/NtAuthTests.Windows.cs

```diff
@@ -0,0 +1,230 @@
+            while (statusCode.ErrorCode == SecurityStatusPalErrorCode.ContinueNeeded);
+
+            if (statusCode.ErrorCode == SecurityStatusPalErrorCode.OK)
+            {
+                // If authentication succeeded ask Windows about the identity and send it back as custom header.
+                SecurityContextTokenHandle? userContext = null;
+                using SafeDeleteContext securityContext = authContext.GetContext(out SecurityStatusPal statusCodeNew)!;
+                SSPIWrapper.QuerySecurityContextToken(GlobalSSPI.SSPIAuth, securityContext, out userContext);
+                using WindowsIdentity identity = new WindowsIdentity(userContext.DangerousGetHandle(), authContext.ProtocolName);
+
+                authHeader = $"{UserHeaderName}: {identity.Name}\r\n";
+                if (closeConnection)
+                {
+                    authHeader += "Connection: close\r\n";
+                }
+
+                await connection.SendResponseAsync(HttpStatusCode.OK, authHeader, "foo");
+                userContext.Dispose();
+            }
+            else
+            {
+                await connection.SendResponseAsync(HttpStatusCode.Forbidden, "Connection: close\r\n", "boo");
+            }
+        }
+
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(true)]
+        [InlineData(false)]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public async Task DefaultHandler_DefaultCredentials_Success(bool useNtlm)
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Get, uri);
+                    requestMessage.Version = new Version(1, 1);
+
+                    var handler = new HttpClientHandler() { UseDefaultCredentials = true };
+                    using (var client = new HttpClient(handler))
+                    {
+                        HttpResponseMessage response = await client.SendAsync(requestMessage);
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        _output.WriteLine($"Authenticated as {response.Headers.GetValues(NtAuthTests.UserHeaderName).First()}");
+                        Assert.Equal("foo", await response.Content.ReadAsStringAsync());
+                    }
+                },
+                async server =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        Task t = useNtlm ? HandleNtlmAuthenticationRequest(connection) : HandleNegotiateAuthenticationRequest(connection);
+                        await t;
+                    }).ConfigureAwait(false);
+                });
+        }
+    }
+
+    public class InpersonificatedNtAuthTests : IClassFixture<WindowsIdentityFixture>
```

Comment:
Should this be in a separate file?

Also the correct term here is "Impersonated"

Reply:
I can move it. I felt it may not be worth do do that for a single test but perhaps it can grow over time.

Reply:
My preference is a separate file.

Reply:
done.